nautobot/extras/jobs.py

```diff
@@ -986,6 +988,21 @@ def run_job(data, request, job_result_pk, commit=True, *args, **kwargs):
         job_result.save()
         return False
 
+    if hasattr(job.Meta, "soft_time_limit"):
```

Comment:
@glennmatthews is this OK?

Reply:
Logic looks reasonable to me. Does it work well to have the check here in `run_job` or would it be possible/better to put it in `JobResult.enqueue_job` instead?